src/code/z_camera.c

```diff
@@ -5252,19 +5252,19 @@ s32 Camera_Unique9(Camera* camera) {
             rwData->curKeyFrame = &ONEPOINT_CS_INFO(camera)->keyFrames[rwData->curKeyFrameIdx];
             rwData->keyFrameTimer = rwData->curKeyFrame->timerInit;
 
-            if (rwData->curKeyFrame->unk_01 != 0xFF) {
-                if ((rwData->curKeyFrame->unk_01 & 0xF0) == 0x80) {
-                    D_8011D3AC = rwData->curKeyFrame->unk_01 & 0xF;
-                } else if ((rwData->curKeyFrame->unk_01 & 0xF0) == 0xC0) {
-                    Camera_UpdateInterface(
-                        CAM_INTERFACE_FIELD(CAM_LETTERBOX_IGNORE, CAM_HUD_VISIBILITY(rwData->curKeyFrame->unk_01), 0));
-                } else if (camera->player->stateFlags1 & PLAYER_STATE1_27 &&
-                           player->currentBoots != PLAYER_BOOTS_IRON) {
+            if (rwData->curKeyFrame->field != ONEPOINT_CS_FIELD_TYPE_NONE) {
+                if ((rwData->curKeyFrame->field & ONEPOINT_CS_FIELD_TYPE_MASK) == ONEPOINT_CS_FIELD_TYPE_ACTORCAT) {
```

Comment:
I'm wondering about
```c
#define ONEPOINT_CS_FIELD_IS_TYPE_ACTORCAT(field) \
((field & ONEPOINT_CS_FIELD_TYPE_MASK) == ONEPOINT_CS_FIELD_TYPE_ACTORCAT)
```

```suggestion
                if (ONEPOINT_CS_FIELD_IS_TYPE_ACTORCAT(rwData->curKeyFrame->field)) {
```

since the bit layout of this field is indeed complicated

hud visibility could have a similar macro, and the "csMode" one which I expect would go unused could in theory be
```c
#define ONEPOINT_CS_FIELD_IS_TYPE_PLAYER_CS(field)   \
(   (field != ONEPOINT_CS_FIELD_TYPE_NONE) \
 && !ONEPOINT_CS_FIELD_IS_TYPE_HUD_VISIBILITY(field) \
 && !ONEPOINT_CS_FIELD_IS_TYPE_ACTORCAT(field))
```
or just &0x80 (bleh)

--------------

An idea:

```c
#define ONEPOINT_CS_FIELD_NONE 0xFF
#define ONEPOINT_CS_FIELD_ACTORCAT(actorCat) \
    (0x80 | ((actorCat) & 0x0F))
#define ONEPOINT_CS_FIELD_HUD_VISIBILITY(camHudVisibility) \
    (0xC0 | ((camHudVisibility) & 0x0F))
#define ONEPOINT_CS_FIELD_PLAYER_CS(csMode) \
    ((csMode) & 0x7F)

#define ONEPOINT_CS_FIELD_IS_TYPE_ACTORCAT(field) \
    ((field & 0xF0) == 0x80)
#define ONEPOINT_CS_FIELD_IS_TYPE_HUD_VISIBILITY(field) \
    ((field & 0xF0) == 0xC0)
#define ONEPOINT_CS_FIELD_IS_TYPE_PLAYER_CS(field) \
    !(field & 0x80)
```

then e.g. `ONEPOINT_CS_FIELD_PLAYER_CS(PLAYER_CSMODE_7)`, `ONEPOINT_CS_FIELD_ACTORCAT(ACTORCAT_BG)` ...

Reply:
That last idea does seem appealing to me, that was what I was roughly thinking with "Alternatively, maybe each data type can have its own variant of a command."